e2e/teams.spec.ts

```diff
@@ -0,0 +1,211 @@
+import { expect, test } from '@playwright/test'
+import { v4 as uuidv4 } from 'uuid'
+import { testAuth } from './fixtures/authenticatedUserPages'
+import {
+  deleteTeamInvitationMessage,
+  deleteTeamInviteAcceptanceMessage,
+} from './helpers/messages'
+import {
+  getAccessTokenFromCookie,
+  getAdminPAT,
+  getUserIdFromLocalStorage,
+  goToDashboard,
+} from './helpers/testUser'
+import {
+  userConfigs,
+  userPrefix,
+  userValidatedPrefix,
+} from './helpers/userConfig'
+
+const TEAM_NAME = 'swc-e2e-team-' + uuidv4()
+const INVITATION_MESSAGE = 'swc-e2e-invite'
+
+// Run multiple describes in parallel, but run tests inside each describe in order
+// ...tests within describe expect afterAll to be run with the same users, i.e. on the same worker
+// https://playwright.dev/docs/api/class-test#test-describe-configure
+test.describe.configure({ mode: 'serial' })
+
+test.describe('Teams', () => {
+  testAuth(
+    'should exercise team lifecycle',
+    async ({ userPage, validatedUserPage }, testInfo) => {
+      const { userName, validatedUserName } = await testAuth.step(
+        'should get user names',
+        async () => {
+          const userName = userConfigs[userPrefix].username
+          const validatedUserName = userConfigs[userValidatedPrefix].username
+          return { userName, validatedUserName }
+        },
+      )
+
+      await testAuth.step(
+        'user should create a team with a unique name',
+        async () => {
+          await goToDashboard(userPage)
+          await userPage.getByLabel('Teams').click()
+          await expect(
+            userPage.getByRole('heading', { name: 'Your Teams' }),
+          ).toBeVisible()
+
+          await userPage
+            .getByRole('button', { name: 'Create a New Team' })
+            .click()
+          const teamNameInput = userPage.getByRole('textbox').nth(1)
+          await teamNameInput.fill(TEAM_NAME)
+          await expect(teamNameInput).toHaveValue(TEAM_NAME)
+
+          await userPage.getByRole('button', { name: 'OK' }).click()
+          await expect(
+            userPage.getByText(`Team Created: ${TEAM_NAME}`),
+          ).toBeVisible()
+        },
+      )
+
+      await testAuth.step(
+        'user should invite validated user to team',
+        async () => {
+          await userPage.getByRole('button', { name: 'Team Actions' }).click()
+          await userPage.getByRole('link', { name: 'Invite User' }).click()
+
+          await userPage
+            .getByRole('textbox', { name: 'Enter a name or email address...' })
+            .pressSequentially(validatedUserName)
```

Comment:
`locator.type` was deprecated in favor of `locator.fill` in [Playwright v1.38.0](https://github.com/microsoft/playwright/releases/tag/v1.38.0). However, user search box requires only finds username when entered letter by letter, so use `locator.pressSequentially` instead.

Reply:
Wonder if this is a bug in UserSearchBox

Reply:
I think the `InviteWidget` is using the `SynapseSuggestBox`, which seems to require a `keydown` event to trigger the users dropdown list -- [`locator.fill`](https://playwright.dev/docs/api/class-locator#locator-fill) only sends an `input` event after filling the input, so the dropdown isn't triggered. However, [`locator.pressSequentially`](https://playwright.dev/docs/api/class-locator#locator-press-sequentially) sends `keydown`, `keypress`/`input`, and `keyup` events for each character in the text, so the dropdown appears.

I'm not sure whether this is an issue that a real user would run into. Should I open a ticket to fix in `SynapseSuggestBox` or consider switching to `UserSearchBoxV2` (which doesn't seem to have the same issue)?

Reply:
It seems to not work in Firefox when I paste using the right-click context menu (pasting using keyboard shortcut works normally). This is probably not a huge deal, but it would be a good idea to create a ticket with these details so we can decide if it's worth the effort later.

Reply:
Good call! Issue created [here](https://sagebionetworks.jira.com/browse/SWC-6569).